dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/folder/FolderResourceTest.java

```diff
@@ -0,0 +1,134 @@
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.folders.business.FolderAPI;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.util.Logger;
+import com.liferay.portal.model.User;
+import com.liferay.util.Base64;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FolderResourceTest {
+
+    static HttpServletResponse response;
+    static FolderResource resource;
+    static FolderAPI folderAPI;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        IntegrationTestInitService.getInstance().init();
+        response = new MockHttpResponse();
+        resource = new FolderResource();
+        folderAPI = APILocator.getFolderAPI();
+    }
+
+    private HttpServletRequest getHttpRequest(final String userEmail,final String password) {
+        final String userEmailAndPassword = userEmail + ":" + password;
+        final MockHeaderRequest request = new MockHeaderRequest(
+                new MockSessionRequest(
+                        new MockAttributeRequest(new MockHttpRequest("localhost", "/").request())
+                                .request())
+                        .request());
+
+            request.setHeader("Authorization",
+                    "Basic " + new String(Base64.encode(userEmailAndPassword.getBytes())));
+
+
+        return request;
+    }
+
+    /**
+     * Method to test: createFolders in the FolderResource
+     * Given Scenario: Create a few folders using the admin as the user
+     * ExpectedResult: The folders should be created successfully, so a 200 code should be returned
+     *
+     */
+    @Test
+    public void test_createFolders_success() throws DotDataException, DotSecurityException {
+        final User adminUser = TestUserUtils.getAdminUser();
+        final long currentTime = System.currentTimeMillis();
+        final List<String> foldersToCreate = Arrays.asList("test_"+currentTime+"/folder_"+currentTime,"/test2_"+currentTime+"/","test3_"+currentTime);
+        final Host newHost = new SiteDataGen().nextPersisted();
+
+        final Response responseResource = resource.createFolders(getHttpRequest(adminUser.getEmailAddress(),"admin"),response,foldersToCreate,newHost.getHostname());
+
+        //Check that the response is 200, OK
+        Assert.assertEquals(Status.OK.getStatusCode(),responseResource.getStatus());
+
+        //Check Results
+        final ResponseEntityView responseEntityView = ResponseEntityView.class.cast(responseResource.getEntity());
+        List<Map<String,Object>> listOfResults = List.class.cast(responseEntityView.getEntity());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'listOfResults' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5552986)